### PR TITLE
Fix focus state in paper-input

### LINKED ISSUE #99: Animation not working with type="number"
The focus animation, floating label animation and other stuff is broken since 1.0.4. I confirmed that it worked in 1.0.3.
See https://github.com/PolymerElements/paper-input/commit/911f56bdc7aef2873e32c9c0b65df7088a023cd9#commitcomment-11789462

### PR DESCRIPTION
**Pull request by morethanreal**
Resolves #99 and also make all tests green. See code comment for more details. PTAL @notwaldorf

### CODE REVIEW

File: paper-input-behavior.html [hunk 1]
```diff
@@ -323,6 +327,24 @@
       return placeholder || alwaysFloatLabel;
     },
 
+    _focusedControlStateChanged: function(focused) {
+      // IronControlState stops the focus and blur events in order to redispatch them on the host
+      // element, but paper-input-container listens to those events. Since there are more
+      // pending work on focus/blur in IronControlState, I'm putting in this hack to get the
+      // input focus state working for now.
+      if (!this.$.container) {
+        this.$.container = Polymer.dom(this.root).querySelector('paper-input-container');
```
Comment (notwaldorf) [inline paper-input-behavior.html]: Whoa, how can there not be a `this.$.container` set. Do i even want to know? :)

Comment (morethanreal) [inline paper-input-behavior.html]: Since other elements outside of this repostory can use this behavior, their templates might not have `container` set.

**Comment by morethanreal** [comment]
@notwaldorf Can I merge this?

**Comment by notwaldorf** [comment]
Yup! lgtm! I'll merge it for you :)

**Pull request merged**